cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -981,10 +982,16 @@ protected void heapOptions(List<EnvVar> envVars, double dynamicHeapFraction, lon
             kafkaHeapOpts.append(' ').append("-Xmx").append(xmx);
         } else {
             ResourceRequirements resources = getResources();
-            Map<String, Quantity> cpuMemory = resources == null ? null : resources.getLimits();
+            Map<String, Quantity> cpuMemoryS = resources == null ? null : resources.getRequests();
+            Map<String, Quantity> cpuMemoryX = resources == null ? null : resources.getLimits();
             // Delegate to the container to figure out only when CGroup memory limits are defined to prevent allocating
             // too much memory on the kubelet.
-            if (cpuMemory != null && cpuMemory.get("memory") != null) {
+            if (cpuMemoryX != null && cpuMemoryX.get("memory") != null) {
+                if (cpuMemoryS != null && cpuMemoryS.get("memory") != null) {
+                    kafkaHeapOpts.append("-Xms").append(Quantities.parseMemory(cpuMemoryS.get("memory").getAmount()));
+                    kafkaHeapOpts.append(" ");
+                }
+                kafkaHeapOpts.append("-Xmx").append(Quantities.parseMemory(cpuMemoryX.get("memory").getAmount()));
```

Comment:
Does it really make sense to enforce the `-Xmx` and set `ENV_VAR_DYNAMIC_HEAP_FRACTION`? From the code it looks like it is one or another.